crates/runtime/src/traphandlers.rs

```diff
@@ -591,45 +592,86 @@ mod tls {
 
     /// Opaque state used to help control TLS state across stack switches for
     /// async support.
-    pub struct TlsRestore {
+    pub struct TlsState {
```

Comment:
Similarly, I wouldn't mind calling this `AsyncWasmCallState` or something along those lines.

Maybe I'm taking this too far...